src/diffusers/quantizers/gguf/utils.py

```diff
@@ -0,0 +1,369 @@
+# Copyright 2024 The HuggingFace Team and City96. All rights reserved.
+# #
+# # Licensed under the Apache License, Version 2.0 (the "License");
+# # you may not use this file except in compliance with the License.
+# # You may obtain a copy of the License at
+# #
+# #     http://www.apache.org/licenses/LICENSE-2.0
+# #
+# # Unless required by applicable law or agreed to in writing, software
+# # distributed under the License is distributed on an "AS IS" BASIS,
+# # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# # See the License for the specific language governing permissions and
+# # limitations under the License.
+
+import gguf
+import torch
+import torch.nn as nn
+
+
+def _replace_with_gguf_linear(model, compute_dtype):
```

Comment:
GGUF files contain a mix of quantized linear and unquantized linear layers. It's not trivial to selectively replace layers. We can replace all of them and then check the parameter type when running forward instead.

Reply:
nvm